chainer/functions/array/spatial_transformer_sampler.py

```diff
@@ -0,0 +1,289 @@
+        self.use_cudnn = use_cudnn
+
+    def check_type_forward(self, in_types):
+        n_in = in_types.size()
+        type_check.expect(2 == n_in)
+
+        x_type = in_types[0]
+        grid_type = in_types[1]
+        type_check.expect(
+            x_type.dtype.kind == 'f',
+            grid_type.dtype.kind == 'f',
+            x_type.ndim == 4,
+            grid_type.ndim == 4,
+            grid_type.shape[1] == 2,
+            x_type.shape[0] == grid_type.shape[0],
+        )
+
+    def forward_cpu(self, inputs):
+        return self._forward(inputs)
+
+    def forward_gpu(self, inputs):
+        if cuda.cudnn_enabled and self.use_cudnn and _cudnn_version >= 5000:
+            x, grid = inputs
+            out_shape = x.shape[:2] + grid.shape[2:]
+            y = cuda.cupy.empty(out_shape, dtype=x.dtype)
+            shape = numpy.array(out_shape, dtype=numpy.int32)
+            x = cuda.cupy.ascontiguousarray(x)
+            grid_t = cuda.cupy.transpose(grid, (0, 2, 3, 1))
+            grid_t = cuda.cupy.ascontiguousarray(grid_t)
+
+            handle = cudnn.get_handle()
+            x_desc = cudnn.create_tensor_descriptor(x)
+            y_desc = cudnn.create_tensor_descriptor(y)
+
+            self.st_desc =\
+                cuda.cupy.cudnn.create_spatial_transformer_descriptor(
+                    _sampler_type, grid.dtype, len(shape), shape.ctypes.data)
+
+            one = numpy.array(1, dtype=x.dtype).ctypes
+            zero = numpy.array(0, dtype=x.dtype).ctypes
+            libcudnn.spatialTfSamplerForward(
+                handle, self.st_desc.value, one.data,
+                x_desc.value, x.data.ptr, grid_t.data.ptr, zero.data,
+                y_desc.value, y.data.ptr)
+            return y,
+        return self._forward(inputs)
+
+    def _forward(self, inputs):
+        x, grid = inputs
+        xp = cuda.get_array_module(x)
+        B, C, H, W = x.shape
+        _, _, out_H, out_W = grid.shape
+
+        grid = grid.reshape(grid.shape[:2] + (-1,))
+
+        u = grid[:, 0]
+        v = grid[:, 1]
+
+        # clip coordinates to [-1, 1]
+        u = u.clip(-1, 1)
+        v = v.clip(-1, 1)
+
+        # rescale coordiantes from [-1, 1] to [0, width or height - 1]
+        u = (u + 1) / 2 * (W - 1)
+        v = (v + 1) / 2 * (H - 1)
+
+        # indices of the 2x2 pixel neighborhood surrounding the coordinates
+        u0 = xp.floor(u).astype(numpy.int32)
+        u0 = u0.clip(0, W - 2)
+        u1 = u0 + 1
+        v0 = xp.floor(v).astype(numpy.int32)
+        v0 = v0.clip(0, H - 2)
+        v1 = v0 + 1
+
+        # weights
+        w1 = (u1 - u) * (v1 - v)
+        w2 = (u - u0) * (v1 - v)
+        w3 = (u1 - u) * (v - v0)
+        w4 = (u - u0) * (v - v0)
+        w1 = w1.astype(x.dtype)
+        w2 = w2.astype(x.dtype)
+        w3 = w3.astype(x.dtype)
+        w4 = w4.astype(x.dtype)
+
+        ys = []
+        for b in range(B):
+            tmp = (w1[b, :, None] * x[b, :, v0[b], u0[b]] +
```

Comment:
`tmp` looks ambiguous as a variable name. How about rename it to another name? (e.g., `elem`, `y`, etc.)